x/programs/cmd/simulator/src/simulator.rs

```diff
@@ -0,0 +1,367 @@
+use crate::codec::{base64_decode, id_from_usize};
+use crate::util::get_path;
+use crate::{param::Param, Id};
+use borsh::BorshDeserialize;
+use serde::{Deserialize, Serialize};
+use std::{
+    io::{BufRead, BufReader, Write},
+    path::Path,
+    process::{Child, Command, Stdio},
+};
+use thiserror::Error;
+use wasmlanche_sdk::ExternalCallError;
+
+/// The endpoint to call for a [`SimulatorRequest`].
+#[derive(Debug, Serialize, PartialEq, Clone)]
+#[serde(rename_all = "lowercase")]
+pub enum Endpoint {
```

Comment:
Is this still being used anywhere outside of this crate? If not, don't leave the `pub` there.